.github/workflows/wheels.yml

```
@@ -24,15 +24,44 @@ env:
   FORCE_COLOR: 1
 
 jobs:
-  macos:
-    uses: ./.github/workflows/wheels-macos.yml
-    with:
-      artifacts-name: "dist"
-
-  linux:
-    uses: ./.github/workflows/wheels-linux.yml
-    with:
-      artifacts-name: "dist"
+  build:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "macOS x86_64"
+            os: macos-latest
+            archs: x86_64
+          - name: "macOS arm64"
+            os: macos-latest
+            archs: arm64
+          - name: "manylinux2014 and musllinux x86_64"
+            os: ubuntu-latest
+            archs: x86_64
+          - name: "manylinux_2_28 x86_64"
+            os: ubuntu-latest
+            archs: x86_64
+            build: "*manylinux*"
+            manylinux: "manylinux_2_28"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.16.2
+        with:
+          output-dir: wheelhouse
+        env:
+          CIBW_ARCHS: ${{ matrix.archs }}
+          CIBW_BUILD: ${{ matrix.build }}
+          CIBW_CONFIG_SETTINGS: raqm=enable raqm=vendor fribidi=vendor
+          CIBW_MANYLINUX_PYPY_X86_64_IMAGE: ${{ matrix.manylinux }}
+          CIBW_MANYLINUX_X86_64_IMAGE: ${{ matrix.manylinux }}
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
```

To put them in the same `dist.zip` as the sdist, rather then the default `artifacts.zip`:

```suggestion
          path: ./wheelhouse/*.whl
          artifacts-name: "dist"
```

---

Good call. I've added a commit to add `name`, as per https://github.com/python-pillow/Pillow/blob/da0b826e960474669f630a419ecc96b8540b52a7/.github/workflows/wheels.yml#L51-L54